src/index/Vocabulary.h

```diff
@@ -89,6 +88,53 @@ class Vocabulary {
   using enable_if_uncompressed =
       std::enable_if_t<!std::is_same_v<T, CompressedString>>;
 
+  static constexpr bool isCompressed_ =
+      std::is_same_v<StringType, CompressedString>;
+
+ private:
+  // If a word starts with one of those prefixes it will be externalized
+  vector<std::string> externalizedPrefixes_;
+
+  // If a word uses one of these language tags it will be internalized,
+  // defaults to English
+  vector<std::string> internalizedLangs_{"en"};
+
+  using PrefixCompressedVocabulary =
+      CompressedVocabulary<VocabularyInMemory, PrefixCompressor>;
+  using InternalCompressedVocabulary =
+      UnicodeVocabulary<PrefixCompressedVocabulary, ComparatorType>;
+  using InternalUncompressedVocabulary =
+      UnicodeVocabulary<VocabularyInMemory, ComparatorType>;
+  using InternalVocabulary =
+      std::conditional_t<isCompressed_, InternalCompressedVocabulary,
+                         InternalUncompressedVocabulary>;
+  InternalVocabulary internalVocabulary_;
+
+  using ExternalVocabulary =
+      UnicodeVocabulary<VocabularyOnDisk, ComparatorType>;
+  ExternalVocabulary externalVocabulary_;
+
+  // Internal type for storing the two ranges of a prefix (in
+  // `internalVocabulary_` and `externalVocabulary_`, respectively).
+  struct PrefixRanges {
+    IndexT rangeInternalBegin_;
+    IndexT rangeInternalEnd_;
+    IndexT rangeExternalBegin_;
+    IndexT rangeExternalEnd_;
+  };
```

Comment:
My idea to unify two of your PRs:
1. Make it public,
2. make the contains function a member of this range.
3. Make this the return type of your getAllPrefixRanges function from the other PR, useful tools have always more than one usage:)

Reply:
Maybe even give it member functions "getInternalRange" and "getExternalRange"
(But maybe only once you integrate it into your other PR).
Then there you always get both of the ranges, but only getting a single range (if required) is possible.

Reply:
That then also reduces the costs of migrating to milestone IDs.

Reply:
I indeed first had a `contains` function as a member of this small class. But then I have to pass `internalVocabulary_` and `externalVocabulary_` as arguments for each of the three calls to `setRangesForPrefix` or when initializing an object of this class. Of course, it doesn't matter performance-wise but that looked very ugly and wrong to me.

Reply:
I don't follow what you want to tell me here?
The `containedInRange` function only depends on the range and the single argument, in fact your current implementation could and should be static. 
Additionally it doesn't interact at all with the `setRangesForPrefix` function, so I don't understand the connection here.
Just to be clear:
The function that gets a prefix and returns the ranges is definitely a member function of the `Vocabulary`. The function that checks whether a given `VocabIndex` is part of some precomputed `PrefixRange` can (and I think should) be a member of the `PrefixRange` class.